br/pkg/lightning/backend/importer/importer_test.go

```diff
@@ -261,33 +264,41 @@ func BenchmarkMutationPool(b *testing.B) {
 	_ = g
 }
 
-func (s *importerSuite) TestCheckTiDBVersion(c *C) {
+func TestCheckTiDBVersion(t *testing.T) {
 	var version string
 	ctx := context.Background()
 
 	mockServer := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		c.Assert(req.URL.Path, Equals, "/status")
+		require.Equal(t, "/status", req.URL.Path)
 		w.WriteHeader(http.StatusOK)
 		err := json.NewEncoder(w).Encode(map[string]interface{}{
 			"version": version,
 		})
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 	}))
 
 	tls := common.NewTLSFromMockServer(mockServer)
 
 	version = "5.7.25-TiDB-v4.0.0"
-	c.Assert(checkTiDBVersionByTLS(ctx, tls, requiredMinTiDBVersion, requiredMaxTiDBVersion), IsNil)
+	require.Nil(t, checkTiDBVersionByTLS(ctx, tls, requiredMinTiDBVersion, requiredMaxTiDBVersion))
 
 	version = "5.7.25-TiDB-v9999.0.0"
-	c.Assert(checkTiDBVersionByTLS(ctx, tls, requiredMinTiDBVersion, requiredMaxTiDBVersion), ErrorMatches, "TiDB version too new.*")
+	err := checkTiDBVersionByTLS(ctx, tls, requiredMinTiDBVersion, requiredMaxTiDBVersion)
+	require.Error(t, err)
+	require.Regexp(t, "TiDB version too new.*", err.Error())
```

Comment:
🤔 

actually since `require.Regexp` uses [`FindStringIndex` with the original pattern](https://github.com/stretchr/testify/blob/ab6dc3262822ed562480c19876b0257ace761e3e/assert/assertions.go#L1408) which passes as long as it matches some substring, in contrast with [`pingcap/check` which adds a `^…$` behind the scene](https://github.com/pingcap/check/blob/57bd13f7b5f0d72a58e63ae7a880e0349c76a426/checkers.go#L373) thus requiring the pattern to match the whole string, this check should really be written as:

```suggestion
	require.Regexp(t, "^TiDB version too new", err.Error())
```

i think we need to review every use of `require.Regexp` added earlier 😓 @tisonkun.

Reply:
It seems that we can fix these mistakes with the help of editor's "search and replace" feature.
1. Add `^..$` to every pattern in `require.Regexp`.
2. Remove the beginning `^.*` or replace the beginning `^^` with `^`.
3. Remove the ending `.*$` or replace the ending `$$` with `$`.

A draft PR for this fix is https://github.com/pingcap/tidb/pull/30080. But we still need to review every modification carefully.

Reply:
So it means I  don’t need to modify this pr like this.

Reply:
@JayL-zxl I would be great if you fixed these mistakes in this PR. Comparing to https://github.com/pingcap/tidb/pull/30080,  it's easier to review.

Reply:
i see  @gozssky

Reply:
@gozssky PTAL again thks